internal/controller/controlplane/k0s_controlplane_controller.go

```diff
@@ -406,6 +406,10 @@ func (c *K0sController) reconcileMachines(ctx context.Context, cluster *clusterv
 			return replicasToReport, fmt.Errorf("error marking controlnode to leave: %w", err)
 		}
 
+		if err := c.deleteControlNode(ctx, name, kubeClient); err != nil {
```

Comment:
@makhov I dont know which logic reacts to this https://github.com/k0sproject/k0smotron/blob/main/internal/controller/controlplane/helper.go#L182-L189 so I thought is safe to remove controlnode at this point. WDYT?

Reply:
No, I think it's too early, let's do it at the very end, after the machine object is deleted

Reply:
done